### DISCUSSION TITLE
$ret better changed to msgbus.system.reply locally instead of on remote side

### DISCUSSION THREAD

**Discussion by sameh-farouk** [threefoldtech/rmb-rs#4] [General]
> [](https://github.com/threefoldtech/rmb-rs/tree/development/docs#rmb-remote-again)RMB (remote)
On receiving a message on the /rmb-remote entrypoint. This entry-point will verify message body, and signature. A 202 Accepted is returned if all is okay, 400 BadRequest, or 401 Unauthorized if signature verifications fails.
If message $dst == $local, $ret is set to msgbus.system.reply and pushed to msgbus.$cmd

We used to change the $ret on the local RMB after storing a copy of the msg on the $backlog and before sending it to its destination.
now I see this done unnecessarily on the remote side.
what is the benefit of sending this info to the remote side instead of handling this locally first?
this info won't be useful for either remote RMB or for the process that handling these messages.

**Comment by muhamadazmy** [comment]
Yup, this can be totally done on the local side before forwarding to remote rmb. BUT The reason I moved this to the remote side is to allow the remote twin to have it's own implementation of how it wants to receive responses. So another rmb (version) or implementation can do something else instead of usign msgbus.system.reply hence also setting this value on local rmb does not make sense. 

IMHO, it's the one that the receive the request that needs to decide which queue it wants to receive responses on basically. Not the sender.

If I send you a letter, I can't tell you how you should handle, or how to type the answer as long as i eventually receive the response I need. While both ways in RMB will have the same effect, what i suggest makes more sense (at least to me)
